Database migration fails

With this error
```
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m: Jan 15, 2026 10:39:55 AM org.flywaydb.core.internal.command.DbMigrate error
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m: SEVERE: Migration of schema "mds_edc_schema" to version "1.1.0 - Lease Fix" failed! Changes successfully rolled back.
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m: SEVERE 2026-01-15T10:39:55.771484647 Error booting runtime: Failed to execute script V1_1_0__Lease_Fix.sql ---------------------------------------------- SQL State  : 23502 Error Code : 0 Message    : ERROR: column "resource_id" of relation "edc_lease" contains null values Location   : migrations/connector/V1_1_0__Lease_Fix.sql (/app/file:/app/lib/database-schema-migration-connector-1.0.0-rc.10.jar!/migrations/connector/V1_1_0__Lease_Fix.sql) Line       : 9 Statement  : Run Flyway with -X option to see the actual statement causing the problem 
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m: org.flywaydb.core.internal.exception.FlywayMigrateException: Failed to execute script V1_1_0__Lease_Fix.sql
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m: ----------------------------------------------
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m: SQL State  : 23502
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m: Error Code : 0
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m: Message    : ERROR: column "resource_id" of relation "edc_lease" contains null values
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m: Location   : migrations/connector/V1_1_0__Lease_Fix.sql (/app/file:/app/lib/database-schema-migration-connector-1.0.0-rc.10.jar!/migrations/connector/V1_1_0__Lease_Fix.sql)
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m: Line       : 9
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m: Statement  : Run Flyway with -X option to see the actual statement causing the problem
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m:   at org.flywaydb.core.internal.command.DbMigrate.doMigrateGroup(DbMigrate.java:399)
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m:   at org.flywaydb.core.internal.command.DbMigrate.lambda$applyMigrations$1(DbMigrate.java:283)
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m:   at org.flywaydb.core.internal.jdbc.TransactionalExecutionTemplate.execute(TransactionalExecutionTemplate.java:59)
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m:   at org.flywaydb.core.internal.command.DbMigrate.applyMigrations(DbMigrate.java:282)
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m:   at org.flywaydb.core.internal.command.DbMigrate.migrateGroup(DbMigrate.java:255)
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m:   at org.flywaydb.core.internal.command.DbMigrate.lambda$migrateAll$0(DbMigrate.java:153)
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m:   at org.flywaydb.database.postgresql.PostgreSQLAdvisoryLockTemplate.execute(PostgreSQLAdvisoryLockTemplate.java:77)
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m:   at org.flywaydb.database.postgresql.PostgreSQLAdvisoryLockTemplate.lambda$execute$0(PostgreSQLAdvisoryLockTemplate.java:60)
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m:   at org.flywaydb.core.internal.jdbc.TransactionalExecutionTemplate.execute(TransactionalExecutionTemplate.java:59)
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m:   at org.flywaydb.database.postgresql.PostgreSQLAdvisoryLockTemplate.execute(PostgreSQLAdvisoryLockTemplate.java:60)
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m:   at org.flywaydb.database.postgresql.PostgreSQLConnection.lock(PostgreSQLConnection.java:105)
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m:   at org.flywaydb.core.internal.schemahistory.JdbcTableSchemaHistory.lock(JdbcTableSchemaHistory.java:164)
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m:   at org.flywaydb.core.internal.command.DbMigrate.migrateAll(DbMigrate.java:153)
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m:   at org.flywaydb.core.internal.command.DbMigrate.migrate(DbMigrate.java:104)
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m:   at org.flywaydb.core.Flyway.lambda$migrate$3(Flyway.java:246)
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m:   at org.flywaydb.core.FlywayExecutor.execute(FlywayExecutor.java:234)
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m:   at org.flywaydb.core.FlywayExecutor.execute(FlywayExecutor.java:118)
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m:   at org.flywaydb.core.Flyway.migrate(Flyway.java:187)
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m:   at eu.dataspace.connector.postgresql.migration.ConnectorPostgresqlMigration.prepare(ConnectorPostgresqlMigration.java:90)
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m:   at org.eclipse.edc.boot.system.injection.lifecycle.ExtensionLifecycleManager.bootServiceExtensions(ExtensionLifecycleManager.java:67)
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m:   at org.eclipse.edc.boot.system.runtime.BaseRuntime.bootExtensions(BaseRuntime.java:160)
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m:   at org.eclipse.edc.boot.system.runtime.BaseRuntime.boot(BaseRuntime.java:104)
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m:   at org.eclipse.edc.boot.system.runtime.BaseRuntime.main(BaseRuntime.java:79)
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m: Caused by: org.flywaydb.core.internal.sqlscript.FlywaySqlScriptException: Failed to execute script V1_1_0__Lease_Fix.sql
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m: ----------------------------------------------
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m: SQL State  : 23502
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m: Error Code : 0
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m: Message    : ERROR: column "resource_id" of relation "edc_lease" contains null values
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m: Location   : migrations/connector/V1_1_0__Lease_Fix.sql (/app/file:/app/lib/database-schema-migration-connector-1.0.0-rc.10.jar!/migrations/connector/V1_1_0__Lease_Fix.sql)
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m: Line       : 9
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m: Statement  : Run Flyway with -X option to see the actual statement causing the problem
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m:   at org.flywaydb.core.internal.sqlscript.DefaultSqlScriptExecutor.handleException(DefaultSqlScriptExecutor.java:251)
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m:   at org.flywaydb.core.internal.sqlscript.DefaultSqlScriptExecutor.executeStatement(DefaultSqlScriptExecutor.java:212)
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m:   at org.flywaydb.core.internal.sqlscript.DefaultSqlScriptExecutor.execute(DefaultSqlScriptExecutor.java:134)
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m:   at org.flywaydb.core.internal.resolver.sql.SqlMigrationExecutor.executeOnce(SqlMigrationExecutor.java:75)
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m:   at org.flywaydb.core.internal.resolver.sql.SqlMigrationExecutor.lambda$execute$0(SqlMigrationExecutor.java:66)
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m:   at org.flywaydb.core.internal.database.DefaultExecutionStrategy.execute(DefaultExecutionStrategy.java:31)
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m:   at org.flywaydb.core.internal.resolver.sql.SqlMigrationExecutor.execute(SqlMigrationExecutor.java:65)
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m:   at org.flywaydb.core.internal.command.DbMigrate.doMigrateGroup(DbMigrate.java:391)
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m:   ... 22 more
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m: Caused by: org.postgresql.util.PSQLException: ERROR: column "resource_id" of relation "edc_lease" contains null values
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m:   at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2736)
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m:   at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2421)
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m:   at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:372)
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m:   at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:525)
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m:   at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:435)
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m:   at org.postgresql.jdbc.PgStatement.executeWithFlags(PgStatement.java:357)
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m:   at org.postgresql.jdbc.PgStatement.executeCachedSql(PgStatement.java:342)
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m:   at org.postgresql.jdbc.PgStatement.executeWithFlags(PgStatement.java:318)
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m:   at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:313)
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m:   at org.flywaydb.core.internal.jdbc.JdbcTemplate.executeStatement(JdbcTemplate.java:229)
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m:   at org.flywaydb.core.internal.sqlscript.ParsedSqlStatement.execute(ParsedSqlStatement.java:88)
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m:   at org.flywaydb.core.internal.sqlscript.DefaultSqlScriptExecutor.executeStatement(DefaultSqlScriptExecutor.java:207)
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m:   ... 28 more
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m: Exception in thread "main" org.eclipse.edc.spi.EdcException: org.flywaydb.core.internal.exception.FlywayMigrateException: Failed to execute script V1_1_0__Lease_Fix.sql
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m: ----------------------------------------------
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m: SQL State  : 23502
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m: Error Code : 0
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m: Message    : ERROR: column "resource_id" of relation "edc_lease" contains null values
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m: Location   : migrations/connector/V1_1_0__Lease_Fix.sql (/app/file:/app/lib/database-schema-migration-connector-1.0.0-rc.10.jar!/migrations/connector/V1_1_0__Lease_Fix.sql)
[Jan 15 2026 11:39:55 GMT+0100] mds-connector-node3-5969dbdff9-k8k5m: Line       : 9
```

--- a/extensions/database-schema-migration-connector/src/main/resources/migrations/connector/V1_0_9__Lease_cleanup.sql
+++ b/extensions/database-schema-migration-connector/src/main/resources/migrations/connector/V1_0_9__Lease_cleanup.sql
@@ -1,0 +1,1 @@
+DELETE FROM edc_lease;

--- a/extensions/database-schema-migration-connector/src/test/java/eu/dataspace/connector/postgresql/migration/ConnectorPostgresqlMigrationTest.java
+++ b/extensions/database-schema-migration-connector/src/test/java/eu/dataspace/connector/postgresql/migration/ConnectorPostgresqlMigrationTest.java
@@ -51,6 +51,7 @@ import java.util.UUID;
 import javax.sql.DataSource;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatNoException;
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
 import static org.flywaydb.core.api.CoreMigrationType.BASELINE;
 import static org.flywaydb.core.api.CoreMigrationType.SQL;
@@ -108,6 +109,27 @@ public class ConnectorPostgresqlMigrationTest {
             assertThat(testMigrationHasBeenApplied(connection)).isEqualTo(true);
         } catch (SQLException e) {
             throw new RuntimeException(e);
+        }
+    }
+
+    @Nested
+    class Lease {
+        @Test
+        void shouldRun110whenLeaseEntryAlreadyExists(ObjectFactory objectFactory, ServiceExtensionContext context) {
+            var participantContextId = UUID.randomUUID().toString();
+            migrateTo(objectFactory, context, "1.0.0", participantContextId);
+
+            try (var connection = createDataSource().getConnection()) {
+                var callableStatement = connection.prepareStatement("insert into edc_lease(leased_by, lease_duration, lease_id) values (?, ?, ?);");
+                callableStatement.setString(1, UUID.randomUUID().toString());
+                callableStatement.setLong(2, 60000L);
+                callableStatement.setString(3, UUID.randomUUID().toString());
+                callableStatement.execute();
+            } catch (SQLException e) {
+                throw new RuntimeException(e);
+            }
+
+            assertThatNoException().isThrownBy(() -> migrateTo(objectFactory, context, "latest", participantContextId));
         }
     }
 

